libraries/core_libs/network/rpc/Taraxa.cpp

```diff
@@ -110,6 +110,52 @@ Json::Value Taraxa::taraxa_getScheduleBlockByPeriod(const std::string& _period)
   }
 }
 
+Json::Value Taraxa::taraxa_getNodeVersions() {
+  try {
+    Json::Value res;
+    auto node = tryGetNode();
+    auto db = node->getDB();
+    auto period = node->getFinalChain()->last_block_number();
+    const uint64_t max_blocks_to_process = 6000;
+    std::map<addr_t, std::string> node_version_map;
+    std::multimap<std::string, addr_t> version_node_map;
+    std::map<std::string, uint32_t> version_count;
+    for (uint64_t i = period; i > 0 && period - i < max_blocks_to_process; i--) {
+      auto blk = db->getPbftBlock(i);
+      if (!blk.has_value()) {
+        break;
+      }
+      if (!node_version_map.contains(blk->getBeneficiary())) {
+        node_version_map[blk->getBeneficiary()] = blk->getExtraData()->getJson()["major_version"].asString() + "." +
+                                                  blk->getExtraData()->getJson()["minor_version"].asString() + "." +
+                                                  blk->getExtraData()->getJson()["patch_version"].asString();
+      }
+    }
+
+    for (auto nv : node_version_map) {
+      version_node_map.insert({nv.second, nv.first});
+      version_count[nv.second]++;
+    }
+
+    res["nodes"] = Json::Value(Json::arrayValue);
+    for (auto vn : version_node_map) {
+      Json::Value node_json;
+      node_json["node"] = vn.second.toString();
+      node_json["version"] = vn.first;
+      res["nodes"].append(node_json);
+    }
+    res["versions"] = Json::Value(Json::arrayValue);
```

Comment:
Aren't we more interested in the votes count and threshold? Maybe it should be added as separate mapping, so which version have how much votes

Reply:
I added vote count for each node and version as well